### PR TITLE
Make fd, ps and SPSPA functions public for the new return system

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**

The new gradient functions become public and the old one become private.

### CODE REVIEW

**Comment by timmysilv** [comment]
I compared the docs for these 3 gradient methods now that the new ones are public-facing (and being rendered), so here are my notes:
- Each transform's "Returns" blurb says: "The type of the matrix returned is either a tensor..." - the word "matrix" should be replaced with something more generic since it's not always a matrix returned
- Each transform's docstring says "This function uses the new return types system" near the beginning, but that can now probably be removed
- Parameter shift usage detail has a line of code that doesn't render properly in the middle of the `broadcast=True` section that starts like "The postprocessing function will know" (this was broken before as well, but might as well fix it)
- The grammar in the SPSA grad blurb comparing it to parameter shift seems a little off, but also.. the old blurb isn't making sense to me either. So I'm not sure exactly what to do to fix that.. let's just review that part of the docs

**Comment by rmoyard** [comment]
@timmysilv Thanks a lot, I answered all your comments expect the grammar of SPSA. I would suggest to come back to that in a separate PR.

File: pennylane/gradients/parameter_shift.py [hunk 1]
```diff
@@ -181,12 +181,9 @@ def _multi_meas_grad(res, coeffs, r0, unshifted_coeff, num_measurements):
     return tuple(g)
 
 
-def _evaluate_gradient_new(tape, res, data, r0, shots):
+def _evaluate_gradient(tape, res, data, r0, shots):
     """Use shifted tape evaluations and parameter-shift rule coefficients to evaluate
     a gradient result. If res is an empty list, ``r0`` and ``data[3]``, which is the
```
Comment (timmysilv) [inline pennylane/gradients/parameter_shift.py:186]:
```suggestion
    a gradient result. If res is an empty list, ``r0`` and ``data[3]``, which is the
    coefficient for the unshifted term, must be given and not None.
```

File: pennylane/gradients/parameter_shift.py [hunk 2]
```diff
@@ -623,22 +625,16 @@ def expval_param_shift(
             saving a quantum evaluation.
         broadcast (bool): Whether or not to use parameter broadcasting to create the
             a single broadcasted tape per operation instead of one tape per shift angle.
-        shots (None, int, list[int]): The device shots that will be used to execute the tapes outputted by this
-            transform. Note that this argument doesn't influence the shots used for tape execution, but provides information
-            to the transform about the device shots and helps in determining if a shot sequence was used to define the
-            device shots for the new return types output system.
+        shots (None, int, list[int]): The device shots that will be used to execute the tapes
+                outputted by this transform. Note that this argument doesn't influence the shots used for tape
+                execution, but provides information to the transform about the shots.
```
Comment (timmysilv) [inline pennylane/gradients/parameter_shift.py:630]:
```suggestion
            outputted by this transform. Note that this argument doesn't influence the shots used for tape
            execution, but provides information to the transform about the shots.
```

File: pennylane/gradients/parameter_shift.py [hunk 3]
```diff
@@ -1060,19 +1059,16 @@ def var_param_shift(
...
```
Comment (timmysilv) [inline pennylane/gradients/parameter_shift.py:1064]:
```suggestion
            outputted by this transform. Note that this argument doesn't influence the shots used for tape
            execution, but provides information to the transform about the shots.
```

File: pennylane/gradients/parameter_shift.py [hunk 4]
```diff
@@ -1660,10 +1665,9 @@ def param_shift(
             saving a quantum evaluation.
         broadcast (bool): Whether or not to use parameter broadcasting to create the
             a single broadcasted tape per operation instead of one tape per shift angle.
-        shots (None, int, list[int]): Argument used by the new return type system (see :func:`~.enable_return` for more
-            information); it represents the device shots that will be used to execute the tapes outputted by this transform.
-            Note that this argument doesn't influence the shots used for tape execution, but provides information to the
-            transform about the device shots and helps in determining if a shot sequence was used.
+        shots (None, int, list[int]): The device shots that will be used to execute the tapes
+                outputted by this transform. Note that this argument doesn't influence the shots used for tape
+                execution, but provides information to the transform about the shots.
```
Comment (timmysilv) [inline pennylane/gradients/parameter_shift.py:1670]:
```suggestion
            outputted by this transform. Note that this argument doesn't influence the shots used for tape
            execution, but provides information to the transform about the shots.
```

**Review by timmysilv — APPROVED** [review]
thanks for the quick swap!

**Review by albi3ro — APPROVED** [review]
Looks good to me 👍

**Pull request merged**